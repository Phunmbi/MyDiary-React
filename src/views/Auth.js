import React, { Component } from 'react';
import { connect } from 'react-redux';
import { testAction } from '../actions/testAction';

export class Auth extends Component {
  handleClick = () => {
    const { testAction } = this.props;
    testAction();
  }

  componentDidUpdate = () => {
    const { response } = this.props;
    console.log(response)

    const result = document.getElementById('result');
    result.innerText = response.middleName;
  }

  render() {
    return (
      <div className="hero">
        <h2>Sign up</h2>
        <p onClick={this.handleClick} className="click">Click me</p>
        <div id="result"></div>
      </div>
    );
  }
};

function mapStatetoProps(state) {
  return { response: state.testReducer.test };
}

export default connect(mapStatetoProps, { testAction })(Auth);
