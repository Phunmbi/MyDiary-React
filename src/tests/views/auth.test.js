import React from 'react';
import Auth from '../../views/Auth';
import Root from '../../root';

const wrapper = shallow(
  <Root>
    <Auth debug />
  </Root>
)

describe('Auth UI', () => {
  it('should render correctly in "debug" mode', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
