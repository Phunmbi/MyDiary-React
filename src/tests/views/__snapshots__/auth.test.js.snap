// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup UI should render correctly in "debug" mode 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Connect(Signup) />
</Provider>
`;
